lib/icinga/downtime.cpp

```diff
@@ -501,7 +549,7 @@ void Downtime::DowntimesExpireTimerHandler()
 {
 	for (const Downtime::Ptr& downtime : ConfigType::GetObjectsByType<Downtime>()) {
 		/* Only remove downtimes which are activated after daemon start. */
-		if (downtime->IsActive() && (downtime->IsExpired() || !downtime->HasValidConfigOwner()))
+		if (downtime->IsActive() && !downtime->HasValidConfigOwner())
```

Comment:
The surrounding function is now misnamed. It doesn't do anything related to expiration any more. Also, wouldn't it be possible to replace the `downtime->HasValidConfigOwner()` check here with something triggered automatically on deletion of ScheduledDowntime objects (similar to how deleting a host can also delete the its services)?

Reply:
Could we **please** not try to solve all problems at one?